### PR TITLE
feat: allow user to manually update manifest file

### PR DESCRIPTION
**Pull request by HuihuiWu-Microsoft**
User can update manifest file manually from editor context menu/preview file codelens/command palette.
For the former two scenarios, update current manifest. For  command palette, ask user to select env to update
update manifest

### CODE REVIEW

**Comment by HuihuiWu-Microsoft** [comment]
E2E TEST: only for extension UI

File: packages/vscode-extension/src/codeLensProvider.ts [hunk 1]
```diff
@@ -149,4 +157,15 @@ export class ManifestTemplateCodeLensProvider implements vscode.CodeLensProvider
 
     return codeLenses;
   }
+
+  private computePreviewCodeLenses(document: vscode.TextDocument) {
+    const codeLenses: vscode.CodeLens[] = [];
+    const command = {
+      title: "🔄Update",
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/codeLensProvider.ts:164]: Will update the icon after UX confirmed

File: packages/fx-core/src/plugins/resource/appstudio/plugin.ts [hunk 2]
```diff
@@ -443,6 +444,124 @@ export class AppStudioPluginImpl {
...
+      }
+    }
+
+    const appStudioToken = await ctx?.appStudioToken?.getAccessToken();
+    const result = await this.updateApp(
+      ctx,
+      appDefinition,
+      appStudioToken!,
+      isLocalDebug,
+      !isLocalDebug,
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/resource/appstudio/plugin.ts:547]: This param is indicating if toolkit will create a new teams app.

If here is false, and the app doesn't exist(e.g. user manually delete it), will cause some 404 error.

Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts:547]: Yes, I'm thinking whether I should create the app for local debug if user haven't done local debug before. Maybe I can confirm with PM.

Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts]: updated

**Review by nliu-ms — APPROVED** [review]

**Comment by HuihuiWu-Microsoft** [comment]
Verified the spfx case, it can be updated successfully if user manually changed it to another provisioned id.

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**